src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,459 @@
+            }
+
+            if (attributes != null)
+            {
+                foreach (AttributeData attribute in attributes)
+                {
+                    if (PlatformAttributeInfo.TryParseAttributeData(attribute, out PlatformAttributeInfo parsedAttribute))
+                    {
+                        if (!(IsSuppressedByTfm(parsedTfms, parsedAttribute) || IsSuppressedByAttribute(parsedAttribute, context.ContainingSymbol)))
+                        {
+                            builder.Add(parsedAttribute);
+                        }
+                    }
+                }
+            }
+
+            if (builder.Count > 0)
+            {
+                platformSpecificOperations.Add(operation, builder.ToImmutable());
+            }
+        }
+
+        private static bool IsWithinConditionalOperation(IOperation pOperation) =>
+         pOperation.Parent is IBinaryOperation bo &&
+            (bo.OperatorKind == BinaryOperatorKind.Equals ||
+            bo.OperatorKind == BinaryOperatorKind.NotEquals ||
+            bo.OperatorKind == BinaryOperatorKind.GreaterThan ||
+            bo.OperatorKind == BinaryOperatorKind.LessThan ||
+            bo.OperatorKind == BinaryOperatorKind.GreaterThanOrEqual ||
+            bo.OperatorKind == BinaryOperatorKind.LessThanOrEqual);
+
+        private static bool IsSuppressedByTfm(List<PlatformAttributeInfo>? parsedTfms, PlatformAttributeInfo parsedAttribute)
+        {
+            if (parsedTfms != null)
+            {
+                foreach (PlatformAttributeInfo tfm in parsedTfms)
+                {
+                    if (IsOSPlatformsEqual(parsedAttribute.OsPlatformName, tfm.OsPlatformName))
+                    {
+                        return AttributeVersionsMatch(parsedAttribute, tfm);
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        private static bool IsOSPlatformsEqual(string firstOs, string secondOs)
+        {
+            return firstOs.Equals(secondOs, StringComparison.OrdinalIgnoreCase)
+                || (firstOs.Equals("OSX", StringComparison.OrdinalIgnoreCase) && secondOs.Equals("MACOS", StringComparison.OrdinalIgnoreCase))
+                || (secondOs.Equals("OSX", StringComparison.OrdinalIgnoreCase) && firstOs.Equals("MACOS", StringComparison.OrdinalIgnoreCase));
+        }
+
+        private static DiagnosticDescriptor SwitchRule(PlatformAttributeType attributeType)
+        {
+            if (attributeType == PlatformAttributeType.MinimumOSPlatformAttribute)
+                return MinimumOsRule;
+            if (attributeType == PlatformAttributeType.ObsoletedInOSPlatformAttribute)
+                return ObsoleteRule;
+            return RemovedRule;
+        }
+
+        private static List<PlatformAttributeInfo>? ParseTfm(AnalyzerOptions options, Compilation compilation, CancellationToken cancellationToken)
```

Comment:
Why is this needed? I believe the analyzer should only look at the assembly/module level `MinimumOSPlatformAttribute`, which is already handled by the analysis.

Reply:
Attributes could be guarded by TFM, i guess this requirement is old now